CONTRIBUTING.md

```diff
@@ -0,0 +1,402 @@
+
+
+## <a name="submit"></a> Submission Guidelines
+
+
+### <a name="submit-issue"></a> Submitting an Issue
+
+Before you submit an issue, please search the [issue tracker][issue-tracker]. An issue for your problem might already exist and the discussion might inform you of workarounds readily available.
+
+Please include the following with each issue:
+
+- Version of Logseq
+- Your operating system
+- List of extensions that you have installed. Attempt to recreate the issue after disabling all extensions.
+- Reproducible steps (1... 2... 3...) that cause the issue
+- What you expected to see, versus what you actually saw
+- Images, animations, or a link to a video showing the issue occurring
+- A code snippet that demonstrates the issue or a link to a  code repository the developers can easily pull down to recreate the  issue locally
+  - **Note:** Because the developers need to copy and paste the code snippet, including a code snippet as a media file (i.e. .gif)  is not sufficient.
+- Errors from the Dev Tools Console (open from the menu: View > Toggle Developer Tools or press CTRL + Shift + i)
+
+You can file new issues by selecting from our [new issue templates][new-issue] and filling out the issue template.
+
+
+### <a name="submit-pr"></a> Submitting a Pull Request (PR)
+
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+1. Search [GitHub][search-pr] for an open or closed PR that relates to your submission.
+   You don't want to duplicate existing efforts.
+
+2. Be sure that an issue describes the problem you're fixing, or documents the design for the feature you'd like to add.
+   Discussing the design upfront helps to ensure that we're ready to accept your work.
+
+3. Please sign our [Contributor License Agreement (CLA)](#cla) before sending PRs.
+   We cannot accept code without a signed CLA.
+   Make sure you author all contributed Git commits with email address associated with your CLA signature.
+
+4. [Fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) the logseq/logseq repo.
+
+5. In your forked repository, make your changes in a new git branch:
+
+     ```shell
+     git checkout -b my-fix-branch master
+     ```
+
+6. Create your patch, **including appropriate test cases**.
+
+7. Follow our [Coding Rules](#rules).
+
+8. Run the full Logseq test suite, as described in the [developer documentation][dev-doc], and ensure that all tests pass.
+
+9. Commit your changes using a descriptive commit message that follows our [commit message conventions](#commit).
+   Adherence to these conventions is necessary because release notes are automatically generated from these messages.
+
+     ```shell
+     git commit --all
+     ```
+    Note: the optional commit `-a` command line option will automatically "add" and "rm" edited files.
+
+10. Push your branch to GitHub:
+
+    ```shell
+    git push origin my-fix-branch
+    ```
+
+11. In GitHub, send a pull request to `logseq:master`.
```

Comment:
It should be mentioned here that Pull Request titles will be used in the changelog, so try to be as descriptive as possible, and that a moderator may edit the title of your pull request.

Reply:
@bendyorke good point. Hopefully with this document, we can make the naming and taging process clear. 

I have been editing the PR titles based on tribal knowledge (tacit knowledge). Having it written down will make it easier for everyone.